### ISSUE TITLE
Bug: [no-var-requires] false positive on TemplateLiteral

### ISSUE THREAD

**Issue by arka1002** [typescript-eslint/typescript-eslint#8994]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://stackblitz.com/edit/stackblitz-starters-qufrpo?file=src%2Findex.ts

### Repro Code

```TypeScript
let packageName = 'foo';
let req = require(`${packageName}/package.json`); // shouldn't be an error

let foo = require('foo/package.json'); // no error, expected behaviour as per tests
```

### ESLint Config

```javascript
/* eslint-env node */
module.exports = {
  parser: '@typescript-eslint/parser',
  plugins: ['@typescript-eslint'],
  parserOptions: {
    project: ['./tsconfig.json'],
    tsconfigRootDir: __dirname,
  },
  root: true,
  ignorePatterns: ['dist/**/*', 'node_modules/**/*'],
  rules: {
    '@typescript-eslint/no-var-requires': [
      'error',
      {
        allow: ['/package\\.json$'],
      },
    ],
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "module": "NodeNext",
    "esModuleInterop": true,
    "target": "ES6",
    "moduleResolution": "NodeNext",
    "sourceMap": true,
    "outDir": "dist",
    "noEmitOnError": true,
    "noImplicitAny": true,
    "strictNullChecks": true
  },
  "lib": ["es2015"],
  "include": ["src/**/*.ts"],
  "exclude": ["dist/**/*"]
}
```

### Expected Result

I expected line 2, in src/index.ts, to not report

### Actual Result

There is an error in line 2, in src/index.ts,

### Additional Info

Steps to reproduce -
1. please go to `src/index.ts` in the stackblitz link
2. run `npx eslint src/index.ts` to see the results

**Comment by arka1002** [comment]
But .... I think string interpolation is a thing which is beyond the boundary of what linters can do ... maybe the PR which shall resolve this should just handle the [null value](https://github.com/typescript-eslint/typescript-eslint/blob/4bed24d8d37ab066b6f3b988fd1ca7accec931c0/packages/eslint-plugin/src/util/getStaticStringValue.ts#L13)

**Comment by bradzacher** [comment]
I don't think this is something that's we want to handle. If you step into dynamic code then you opt out of our static checks.

Same as if you did `require(a + "package.json")` or `require(path.join(a, "package.json"))`.

**Comment by Josh-Cena** [comment]
Yeah, I think it's better to report it by default. The same way you may have `require(something)` where you know at runtime `something` is a JSON file. The `allow` option is only meant as a best-effort to save some ignore comments.

**Comment by bradzacher** [comment]
Furthermore - technically the `path.join` version is better because portable paths - so if we support this template literal style then we would endorse a sub-par pattern!

Looks like we're in agreement - closing
